core/net/ip/ip64-addr.c

```diff
@@ -36,6 +36,61 @@
 
 #define printf(...)
 /*---------------------------------------------------------------------------*/
+static int
+is_private_ipv4addr(const uip_ip4addr_t *ipv4addr)
+{
+  uip_ipaddr_t ipaddr, mask;
+
+  /* 10.0.0.0/8 */
+  uip_ipaddr(&ipaddr, 10, 0, 0, 0);
+  uip_ipaddr(&mask, 255, 0, 0, 0);
+  if(uip_ipaddr_maskcmp(ipv4addr, &ipaddr, &mask)) {
+    return 1;
+  }
+
+  /* 172.16.0.0/12 */
+  uip_ipaddr(&ipaddr, 172, 16, 0, 0);
+  uip_ipaddr(&mask, 255, 240, 0, 0);
+  if(uip_ipaddr_maskcmp(ipv4addr, &ipaddr, &mask)) {
+    return 1;
+  }
+
+  /* 198.18.0.0/15 */
+  uip_ipaddr(&ipaddr, 198, 18, 0, 0);
+  uip_ipaddr(&mask, 255, 254, 0, 0);
+  if(uip_ipaddr_maskcmp(ipv4addr, &ipaddr, &mask)) {
+    return 1;
+  }
+
+  /* 192.168.0.0/16 */
+  uip_ipaddr(&ipaddr, 192, 168, 0, 0);
+  uip_ipaddr(&mask, 255, 255, 0, 0);
+  if(uip_ipaddr_maskcmp(ipv4addr, &ipaddr, &mask)) {
+    return 1;
+  }
+
+  /* 192.0.2.0/24 */
+  uip_ipaddr(&ipaddr, 192, 0, 2, 0);
+  uip_ipaddr(&mask, 255, 255, 255, 0);
+  if(uip_ipaddr_maskcmp(ipv4addr, &ipaddr, &mask)) {
+    return 1;
+  }
+
+  /* 198.51.100.0/24 */
+  uip_ipaddr(&ipaddr, 198, 51, 100, 0);
```

Comment:
I would add the `uip_ipaddr(&mask, 255, 255, 255, 0);` line here and in the following check as well.
Otherwise these will quickly fail if someone misses to see the link and adds a check in between somewhere.